src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantValidator.cs

```diff
@@ -64,36 +56,63 @@ public async Task<IEnumerable<ModelError>> ValidateAsync(TenantViewModel model)
                 errors.Add(new ModelError(nameof(model.Name), S["Invalid tenant name. Must contain characters only and no spaces."]));
             }
 
-            if (!_shellSettings.IsDefaultShell() && String.IsNullOrWhiteSpace(model.RequestUrlHost) && String.IsNullOrWhiteSpace(model.RequestUrlPrefix))
+            if (!ShellHelper.DefaultShellName.Equals(model.Name, StringComparison.OrdinalIgnoreCase) && String.IsNullOrWhiteSpace(model.RequestUrlHost) && String.IsNullOrWhiteSpace(model.RequestUrlPrefix))
```

Comment:
@jtkech This is the change you request I think. Replacing `_shellSettings.IsDefaultShell()` since it add no value being injected.

Reply:
Yes, but in fact these lines can be just removed as this check is useless as any tenant can have an empty host and prefix, the only requirement is that only one tenant can have this including the Default tenant.

Note: But I agree that it is not recommended that a non Default tenant has both empty host and url prefix, maybe the specifications has changed (needs to be checked).

Reply:
If we want to keep this test (if recommended) but make it working (I tried it, currently we can create a tenant with both empty host and url prefix e.g. if the default tenant has a non empty url prefix) we just need to remove `_shellSettings.IsDefaultShell()` from the if condition.

    if (String.IsNullOrWhiteSpace(model.RequestUrlHost) &&
        String.IsNullOrWhiteSpace(model.RequestUrlPrefix))

Reply:
I think that rule exists just to reserve the “catch all/localhost” to the default-tenant/host. Otherwise, when running locally every time the app runs, we launch a non-default tenant. I am not sure if we should change that logic. I’ll await clear guidance form you or other to change it or leave it. Maybe @hishamco can provide more feedback on that matter since he created the TenantValidator originally.

Reply:
Okay no problem, I'm not against preserving the intended behavior, but if you read my comments this intended behavior is not working.

Note: Yes if the default tenant is not `DefaultIsCatchAll()` we then call `TryMatchInternal("", "", "/", out result)` letting the ability to another tenant to serve on empty host/urlPrefix.

Anyway, tomorrow I will start a new PR to show what I mean and it will be easier for me to let people explain to me why I should not do this or that ;)